src/openqaoa-pyquil/backends/qaoa_pyquil_qpu.py

```diff
@@ -269,12 +269,14 @@ def parametric_qaoa_circuit(self) -> Program:
             for i,qbit in enumerate(self.problem_reg):
                 parametric_circuit += gates.MEASURE(qbit, ro[i])
         else:
+            
             # Measurement instructions
             for qubit in self.qureg:
                 reorderd_qubit = self.final_mapping[qubit]
+
                 if reorderd_qubit in self.problem_reg:
                     cbit = ro[reorderd_qubit]
-                    parametric_circuit += gates.MEASURE(self.qubit_mapping[qubit], cbit)
+                    parametric_circuit += gates.MEASURE(self.initial_qubit_mapping[self.final_mapping[reorderd_qubit]], cbit)
```

Comment:
This piece of code looks a bit unsettling! Why are we indexing `self.final_mapping` again even though the `reordered_qubit` is queried from `self.final_mapping` itself? Would you mind explaining what this piece of code is trying to achieve? Thanks